app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -135,16 +135,30 @@ def stop_middleware_server(ems_ref)
       run_generic_operation(:Shutdown, ems_ref)
     end
 
-    def stop_middleware_deployment(ems_ref)
+    def stop_middleware_deployment(ems_ref, name)
```

Comment:
@pilhuhn Added the undercores.
@abonas Rebased on master to incorporate gem change.
> so this is not fixed yet on hawkular side?

The fix has many touch points.  Requires these changes to MIQ, Hawkular gem changes (already merged) to generate the proper request, and then on the hawkular side there is work going on in agent and inventory to properly sync inventory such that removals are handled correctly.  The Hawkular work does not block this PR but the reported problem won't go away until it is complete.